control-plane/gateways/deployment_init_container.go

```diff
@@ -30,15 +32,15 @@ type initContainerCommandData struct {
 	LogJSON  bool
 }
 
-// containerInit returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
+// initContainer returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
 // so that it can save the proxy service id to the shared volume and boostrap Envoy with the proxy-id.
-func initContainer(config GatewayConfig, name, namespace string) (corev1.Container, error) {
+func (b *meshGatewayBuilder) initContainer() (corev1.Container, error) {
```

Comment:
also, what do you think about changing the name of this method to something like `buildInitContainer` or `newInitContainer`, something to specify that this is creating the init container config rather than initializing a container

Reply:
Yes. I like it. I think I'll do it all at once in another PR though.